app/services/PaymentHistoryService.scala

```diff
@@ -38,40 +38,31 @@ class PaymentHistoryService @Inject()(repaymentHistoryConnector: RepaymentHistor
                                       val appConfig: FrontendAppConfig)
                                      (implicit ec: ExecutionContext) {
 
-  @deprecated("Use getPaymentHistoryV2 instead", "MISUV-8845")
-  def getPaymentHistory(implicit hc: HeaderCarrier, user: MtdItUser[_]): Future[Either[PaymentHistoryError.type, List[Payment]]] = {
-
-    val orderedTaxYears: List[Int] = user.incomeSources.orderedTaxYearsByYearOfMigration.reverse.take(appConfig.paymentHistoryLimit)
-
-    Future.sequence(orderedTaxYears.map(year => financialDetailsConnector.getPayments(TaxYear(year-1, year)))) map { paymentResponses =>
-      val paymentsContainsFailure: Boolean = paymentResponses.exists {
-        case Payments(_) => false
-        case PaymentsError(status, _) if status == NOT_FOUND => false
-        case PaymentsError(_, _) => true
-      }
-      if (paymentsContainsFailure) {
-        Left(PaymentHistoryError)
-      } else {
-        Right(paymentResponses.collect {
-          case Payments(payments) => payments
-        }.flatten.distinct)
-      }
-    }
-  }
-
-  def getPaymentHistoryV2(implicit hc: HeaderCarrier, user: MtdItUser[_]): Future[Either[PaymentHistoryError.type, Seq[Payment]]] = {
+  def getPaymentHistory(implicit hc: HeaderCarrier, user: MtdItUser[_]): Future[Either[PaymentHistoryError.type, Seq[Payment]]] = {
 
     val orderedTaxYears: List[Int] = user.incomeSources.orderedTaxYearsByYearOfMigration.reverse.take(appConfig.paymentHistoryLimit)
 
     val (from, to) = (orderedTaxYears.min, orderedTaxYears.max)
     Logger("application").debug(s"Getting payment history for TaxYears: ${from} - ${to}")
 
-    for {
-      response <- financialDetailsConnector.getPayments(TaxYear.forYearEnd(from), TaxYear.forYearEnd(to))
-    } yield response match {
-      case Payments(payments) => Right(payments.distinct)
-      case PaymentsError(_, _) => Left(PaymentHistoryError)
-    }
+    val maxYears = 5
+    val listOfCalls = orderedTaxYears.grouped(maxYears).toList
```

Comment:
worth to add function which will return list of TaxYearRanges, something like:
user.incomeSources.getOrderedTaxYears[Window/Chunks] or similar